gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -615,15 +615,33 @@ case class TransformPostOverrides(isAdaptiveContext: Boolean) extends Rule[Spark
   def transformColumnarToRowExec(plan: ColumnarToRowExec): SparkPlan = {
     if (columnarConf.enableNativeColumnarToRow) {
       val child = replaceWithTransformerPlan(plan.child)
-      logDebug(s"ColumnarPostOverrides ColumnarToRowExecBase(${child.nodeName})")
-      val nativeConversion =
-        BackendsApiManager.getSparkPlanExecApiInstance.genColumnarToRowExec(child)
-      val validationResult = nativeConversion.doValidate()
-      if (validationResult.isValid) {
-        nativeConversion
-      } else {
-        TransformHints.tagNotTransformable(plan, validationResult)
+      def isChildGlutenPlan(p: SparkPlan): Boolean = {
+        p match {
+          case a: AQEShuffleReadExec => isChildGlutenPlan(a.child)
+          case s: QueryStageExec => isChildGlutenPlan(s.plan)
+          case s: ReusedExchangeExec => isChildGlutenPlan(s.child)
+          case s: InputAdapter => isChildGlutenPlan(s.child)
+          case s: WholeStageCodegenExec => isChildGlutenPlan(s.child)
+          case s: AdaptiveSparkPlanExec => isChildGlutenPlan(s.executedPlan)
+          case _: GlutenPlan => true
```

Comment:
we should also handle columnar table cache

Reply:
Thanks for reviewing, can you point me the plan class name?